extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js

```diff
@@ -0,0 +1,210 @@
+ve.ui.WikiaInfoboxInsertDialog.static.size = 'medium';
+
+/* Methods */
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getSetupProcess = function ( data ) {
+	return ve.ui.WikiaInfoboxInsertDialog.super.prototype.getSetupProcess.call( this, data )
+		.next( function () {
+			this.surface = data.surface;
+		}, this );
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.getBodyHeight = function () {
+	return 600;
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.initialize = function () {
+	// Parent method
+	ve.ui.WikiaInfoboxInsertDialog.super.prototype.initialize.call( this );
+
+	// Initialization
+	this.$content.addClass( 've-ui-wikiaInfoboxInsertDialog' );
+
+	// Load select widget
+	this.loadInfoboxTemplates();
+	this.select = new OO.ui.SelectWidget();
+
+	// Events
+	this.select.connect( this, {
+		select: 'onInfoboxTemplateSelect'
+	} );
+
+	this.$body.append( this.select.$element );
+};
+
+/**
+ * Handle selecting results.
+ *
+ * @method
+ * @param {Object|null} itemData Data of selected item, or null
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.onInfoboxTemplateSelect = function ( itemData ) {
+	var template;
+
+	if ( itemData ) {
+		this.$frame.startThrobbing();
+		this.transclusionModel = new ve.dm.WikiaTransclusionModel();
+		template = ve.dm.MWTemplateModel.newFromName(
+			this.transclusionModel, itemData.data
+		);
+		this.transclusionModel.addPart( template )
+			.done( this.insertInfoboxTemplate.bind( this ) );
+
+		// TODO: Track
+		ve.track( 'wikia', {
+			action: ve.track.actions.ADD,
+			label: 'infobox-template-insert-from-plain-list'
+		} );
+	}
+};
+
+ve.ui.WikiaInfoboxInsertDialog.prototype.loadInfoboxTemplates = function () {
+	this.getInfoboxTemplates().done(
+		this.showResults.bind( this )
+	);
+};
+
+ve.ui.WikiaInfoboxInsertDialog.prototype.getInfoboxTemplates = function () {
+	var deferred;
+	if ( !this.gettingTemplateNames ) {
+		deferred = $.Deferred();
+		$.ajax( {
+			dataType: 'json',
+			//TODO: change to non-mocky url
+			url: 'http://public.diana.wikia-dev.com/infobox_templates.php'
+		} )
+			.done( function ( data ) {
+				deferred.resolve( data.items );
+			} )
+			.fail( function () {
+				// TODO: Add better error handling.
+				deferred.resolve( [] );
+			} );
+		this.gettingTemplateNames = deferred.promise();
+	}
+	return this.gettingTemplateNames;
+};
+
+/**
+ * Insert template
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
+	ve.init.target.constructor.static.apiRequest({
+		action: 'visualeditor',
+		paction: 'parsefragment',
+		page: mw.config.get( 'wgRelevantPageName' ),
+		wikitext: this.transclusionModel.getWikitext()
+	}, { type: 'POST' } )
+		.done( this.onParseSuccess.bind( this ) )
+		.fail( function () {
+		// TODO: Implement some proper handling, at least tracking
+	}.bind( this ) );
+};
+
+ve.ui.WikiaInfoboxInsertDialog.prototype.showResults = function ( data ) {
```

Comment:
In VisualEditor we document briefly all methods - useful especially for params :)